packages/example-log-extension/README.md

```diff
@@ -294,36 +306,51 @@ export class LogActionProvider implements Provider<LogFn> {
       level !== LOG_LEVEL.OFF
     ) {
       if (!args) args = [];
-      let log = `${req.url} :: ${controllerClass.name}.`;
-      log += `${methodName}(${args.join(', ')}) => `;
+      let msg = `${req.url} :: ${controllerClass.name}.`;
+      msg += `${methodName}(${args.join(', ')}) => `;
 
-      if (typeof result === 'object') log += JSON.stringify(result);
-      else log += result;
+      if (typeof result === 'object') msg += JSON.stringify(result);
+      else msg += result;
 
       if (start) {
         const timeDiff: HighResTime = this.timer(start);
         const time: number =
           timeDiff[0] * 1000 + Math.round(timeDiff[1] * 1e-4) / 100;
-        log = `${time}ms: ${log}`;
+        msg = `${time}ms: ${msg}`;
       }
 
-      switch (level) {
-        case LOG_LEVEL.DEBUG:
-          console.log(chalk.white(`DEBUG: ${log}`));
-          break;
-        case LOG_LEVEL.INFO:
-          console.log(chalk.green(`INFO: ${log}`));
-          break;
-        case LOG_LEVEL.WARN:
-          console.log(chalk.yellow(`WARN: ${log}`));
-          break;
-        case LOG_LEVEL.ERROR:
-          console.log(chalk.red(`ERROR: ${log}`));
-          break;
+      let logger: LogFn;
+      try {
+        // Try to see if an external logger is bound
+        // This is a workaround before optional injection is supported
```

Comment:
This may not be necessary if property-based injection (see my comment above) already supports optional injection.

If it does not, then please add a link to the issue or pull request tracking the task of adding support for optional injection.

Reply:
See https://github.com/strongloop/loopback-next/pull/946 for optional injection. I'll rebase to #946 once it's merged.